// yup-locale-ja/es5/descriptive.js
// NOTE: `notType` is omitted because it's too hard to users to understand the
//       concept and what's happening. Write your own if you need that one.
export var mixed = {
    default: function (_a) {
        var label = _a.label;
        return (label ? label + "の" : "") + "値が不正です";
    },
    required: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "必須です";
    },
    oneOf: function (_a) {
        var label = _a.label, values = _a.values;
        return (label ? label + "には" : "") + "\u4EE5\u4E0B\u306E\u3082\u306E\u304C\u5165\u529B\u3067\u304D\u307E\u3059: ".concat(values);
    },
    notOneOf: function (_a) {
        var label = _a.label, values = _a.values;
        return (label ? label + "には" : "") + "\u4EE5\u4E0B\u306E\u3082\u306E\u306F\u5165\u529B\u3067\u304D\u307E\u305B\u3093: ".concat(values);
    },
    // From user's perspective, defined and default is essentially same.
    defined: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "必須です";
    },
};
export var string = {
    length: function (_a) {
        var label = _a.label, length = _a.length;
        return (label ? label + "は" : "") + "".concat(length, "\u6587\u5B57\u3067\u3042\u308B\u5FC5\u8981\u304C\u3042\u308A\u307E\u3059");
    },
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "は" : "") + "".concat(min, "\u6587\u5B57\u4EE5\u4E0A\u306E\u307F\u5165\u529B\u3067\u304D\u307E\u3059");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "は" : "") + "".concat(max, "\u6587\u5B57\u307E\u3067\u5165\u529B\u3067\u304D\u307E\u3059");
    },
    matches: function (_a) {
        var label = _a.label, regex = _a.regex;
        return (label ? label + "は" : "") + "\u4EE5\u4E0B\u306E\u5F62\u5F0F\u3067\u3042\u308B\u5FC5\u8981\u304C\u3042\u308A\u307E\u3059: ".concat(regex);
    },
    email: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "正しいメールアドレスではありません";
    },
    url: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "正しいURLではありません";
    },
    uuid: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "正しいUUIDではありません";
    },
    trim: function (_a) {
        var label = _a.label;
        return (label ? label + "の" : "") + "前後に空白は含められません";
    },
    lowercase: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "小文字のみ入力できます";
    },
    uppercase: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "大文字のみ入力できます";
    },
};
export var number = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "には" : "") + "".concat(min, "\u4EE5\u4E0A\u306E\u6570\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "には" : "") + "".concat(max, "\u4EE5\u4E0B\u306E\u6570\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
    lessThan: function (_a) {
        var label = _a.label, less = _a.less;
        return (label ? label + "には" : "") + "".concat(less, "\u672A\u6E80\u306E\u6570\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
    moreThan: function (_a) {
        var label = _a.label, more = _a.more;
        return (label ? label + "には" : "") + "".concat(more, "\u3088\u308A\u5927\u304D\u3044\u6570\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
    positive: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "正の数のみ入力できます";
    },
    negative: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "負の数のみ入力できます";
    },
    integer: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "整数のみ入力可能です";
    },
};
// To format date, use the second argument of yup.date().min/max.
export var date = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "には" : "") + "".concat(min, "\u4EE5\u964D\u306E\u65E5\u4ED8\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "には" : "") + "".concat(max, "\u4EE5\u524D\u306E\u65E5\u4ED8\u306E\u307F\u5165\u529B\u53EF\u80FD\u3067\u3059");
    },
};
export var object = {
    noUnknown: function (_a) {
        var label = _a.label, unknown = _a.unknown;
        return (label ? label + "には" : "") +
            "登録されていないキーは入力できません" +
            (unknown ? ": " + unknown : "");
    },
};
export var array = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "は" : "") + "".concat(min, "\u3064\u4EE5\u4E0A\u5FC5\u8981\u3067\u3059");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "は" : "") + "".concat(max, "\u3064\u307E\u3067\u5165\u529B\u3067\u304D\u307E\u3059");
    },
    length: function (_a) {
        var label = _a.label, length = _a.length;
        return (label ? label + "は" : "") + "".concat(length, "\u3064\u3067\u3042\u308B\u5FC5\u8981\u304C\u3042\u308A\u307E\u3059");
    },
};
/**
 * Descriptive messages.
 */
export var descriptive = {
    mixed: mixed,
    string: string,
    number: number,
    date: date,
    object: object,
    array: array,
};
