// yup-locale-ja/es5/suggestive.js
// NOTE: `notType` is omitted because it's too hard to users to understand the
//       concept and what's happening. Write your own if you need that one.
export var mixed = {
    default: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "正しい値を入力してください";
    },
    required: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "値を入力してください";
    },
    oneOf: function (_a) {
        var label = _a.label, values = _a.values;
        return (label ? label + "には" : "") + "".concat(values, "\u306E\u3044\u305A\u308C\u304B\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    notOneOf: function (_a) {
        var label = _a.label, values = _a.values;
        return (label ? label + "には" : "") + "".concat(values, "\u4EE5\u5916\u306E\u3082\u306E\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    // From user's perspective, defined and default is essentially same.
    defined: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "値を入力してください";
    },
};
export var string = {
    length: function (_a) {
        var label = _a.label, length = _a.length;
        return (label ? label + "は" : "") + "".concat(length, "\u6587\u5B57\u3067\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "は" : "") + "".concat(min, "\u6587\u5B57\u4EE5\u4E0A\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "は" : "") + "".concat(max, "\u6587\u5B57\u4EE5\u4E0B\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    matches: function (_a) {
        var label = _a.label, regex = _a.regex;
        return (label ? label + "は" : "") + "\u4EE5\u4E0B\u306E\u5F62\u5F0F\u3067\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044: ".concat(regex);
    },
    email: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "正しいメールアドレスを入力してください";
    },
    url: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "正しいURLを入力してください";
    },
    uuid: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "正しいUUIDを入力してください";
    },
    trim: function (_a) {
        var label = _a.label;
        return (label ? label + "の" : "") + "前後の空白を取り除いてください";
    },
    lowercase: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "小文字のみ入力してください";
    },
    uppercase: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "大文字のみ入力してください";
    },
};
export var number = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "は" : "") + "".concat(min, "\u4EE5\u4E0A\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "は" : "") + "".concat(max, "\u4EE5\u4E0B\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    lessThan: function (_a) {
        var label = _a.label, less = _a.less;
        return (label ? label + "は" : "") + "".concat(less, "\u672A\u6E80\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    moreThan: function (_a) {
        var label = _a.label, more = _a.more;
        return (label ? label + "は" : "") + "".concat(more, "\u3088\u308A\u5927\u304D\u3044\u6570\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    positive: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "正の数にしてください";
    },
    negative: function (_a) {
        var label = _a.label;
        return (label ? label + "は" : "") + "負の数にしてください";
    },
    integer: function (_a) {
        var label = _a.label;
        return (label ? label + "には" : "") + "整数を入力してください";
    },
};
// To format date, use the second argument of yup.date().min/max.
export var date = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "には" : "") + "".concat(min, "\u4EE5\u964D\u306E\u65E5\u4ED8\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "には" : "") + "".concat(max, "\u4EE5\u524D\u306E\u65E5\u4ED8\u3092\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
};
export var object = {
    // There is no way to describe `noUnknown` message in suggestive way, right?
    noUnknown: function (_a) {
        var label = _a.label, unknown = _a.unknown;
        return (label ? label + "には" : "") +
            "登録されていないキーは入力できません" +
            (unknown ? ": " + unknown : "");
    },
};
export var array = {
    min: function (_a) {
        var label = _a.label, min = _a.min;
        return (label ? label + "は" : "") + "".concat(min, "\u3064\u4EE5\u4E0A\u5165\u529B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    max: function (_a) {
        var label = _a.label, max = _a.max;
        return (label ? label + "は" : "") + "".concat(max, "\u3064\u4EE5\u4E0B\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
    length: function (_a) {
        var label = _a.label, length = _a.length;
        return (label ? label + "は" : "") + "".concat(length, "\u3064\u306B\u3057\u3066\u304F\u3060\u3055\u3044");
    },
};
/**
 * Suggestive messages.
 */
export var suggestive = {
    mixed: mixed,
    string: string,
    number: number,
    date: date,
    object: object,
    array: array,
};
